Update Checkpoint parser

KGuillemot opened the pull request:
Update Checkpoint parser to parse newest format with quoted strings, for example : 
[action:"Accept"; dst:"192.168.1.2"; product:"FireWall"; src:"192.168.1.3"; ]

KGuillemot committed:
PARSER :: Improvement : Improve Checkpoint parser to accept quoted strings + Tests

KGuillemot commented:
Hello,
Do you know what is going on concerning the tests failures ?

solhuebner commented:
Also, have a look at #234 as it is meant to fix the no quotes support so that we do not try to fix it in different ways

KGuillemot commented:
> Also, have a look at #234 as it is meant to fix the no quotes support so that we do not try to fix it in different ways

Hello,
No, it's a different parser.

rgerhards commented:
Top go forward, the CI failures must be fixed. I guess they indicate a backwards.-incompatible changed happened.

Commit:
FIX CHECKPOINT-LEA: ';' NOT NEEDED AFTER LAST VALUE

As of today, Checkpoint logs do not necessarily provide a semicolon
between the last key:value pair and the terminator.
This causes the parse to fail with current behavior.

solhuebner commented:
Almost forgot this project. Great that it is still alive 👍

rgerhards commented:
> Almost forgot this project. Great that it is still alive 👍

yeah, I am awfully sorry. It worked so reliable for us, and the during Covid I somehow lost attention to it.

I also expect some great new contributions and am working through the backlog.

Thanks for keeping interested!
